### PR TITLE
Rename Proto-related podspecs

### PR DESCRIPTION
**Pull request by muxi**
It seems that Cocoapods 1.3.1 is not supporting exclamation in podspec name, and causing `pod install` to fail (#12172). While we are not sure if this is intended behavior and are contacting Cocoapods, this PR is a workaround we are considering that may resolve this issue by removing the prefix.

### CODE REVIEW

File: src/objective-c/ProtoCompiler-gRPCPlugin.podspec [hunk 1]
```diff
@@ -41,7 +41,7 @@ Pod::Spec.new do |s|
   # Because CocoaPods installs pods in alphabetical order, beginning this pod's name with an
   # exclamation mark ensures that other "regular" pods will be able to find it as it'll be installed
   # before them.
-  s.name     = '!ProtoCompiler-gRPCPlugin'
+  s.name     = 'ProtoCompiler-gRPCPlugin'
```
Comment (jcanizales) [inline src/objective-c/ProtoCompiler-gRPCPlugin.podspec]: You need to check if the restriction documented here for the name still applies.

Comment (derpoliuk) [reply src/objective-c/ProtoCompiler-gRPCPlugin.podspec]: @jcanizales not a contributor, but I'm curios what exactly do you mean here? What is `!` in name for?

**Comment by derpoliuk** [comment]
@muxi is there any response from CocoaPods?

**Comment by dnkoutso** [comment]
Old PR but FWIW this has been fixed in CocoaPods. I believe it has shipped with 1.5.x.

**Comment by muxi** [comment]
Yep we were already using 1.5.0 with no problem. I just forgot to close this PR... Thanks Dmitris.

**Pull request closed without merging**